lightning/src/ln/wire.rs

```diff
@@ -96,6 +105,7 @@ impl Message {
 			&Message::ReplyChannelRange(ref msg) => msg.type_id(),
 			&Message::GossipTimestampFilter(ref msg) => msg.type_id(),
 			&Message::Unknown(type_id) => type_id,
+			&Message::Custom(_) => MessageType(0),
```

Comment:
A message shouldn't have a default type, I think `Custom` needs to store the type somehow.

Reply:
@jkczyz how do you implement `Encode` for an enum with the const requirement?

Reply:
Hmm... good point. I guess it could instead be `Custom(MessageType, T)` or implement `Encode` for the enum as `0` and override `type_id`.

Reply:
With `Custom(MessageType, T)` if I'm not mistaken it would  only allow you to have a single value for `MessageType` so I don't think it's great. The override would probably work but doesn't feel super clean, and it would make the API a bit complicated on the user side I feel.

Reply:
Here `T` is the enum, right? If so, it could take on an variant.

Reply:
I still don't really see what you mean tbh but since it's not relevant with the new approach maybe no need to spend more time on that.

Reply:
So it's back to being relevant and I think I got it (I hope at least).